src/world.rs

```diff
@@ -54,15 +55,13 @@ pub struct World {
 impl World {
     /// Create an empty world
     pub fn new() -> Self {
-        static ID: AtomicU64 = AtomicU64::new(1);
+        static ID: Mutex<u64> = Mutex::new(1);
 
         Self {
             entities: Entities::default(),
             archetypes: ArchetypeSet::new(),
             bundle_to_archetype: HashMap::default(),
-            id: ID
-                .fetch_update(Ordering::Relaxed, Ordering::Relaxed, |id| id.checked_add(1))
-                .unwrap(),
+            id: *ID.lock() + 1,
```

Comment:
This just adds 1 to the current value, but leaves that value unchanged, so every invocation will get the same result. We should compute a new value using `checked_add`, panic if it overflows, store the new value, then yield it for use as `id`.